### DISCUSSION TITLE
Switch SNOWFLAKE_ROLE for resource creation using VCS in Terraform Cloud

### DISCUSSION THREAD

**Discussion by tamathew** [Snowflake-Labs/terraform-provider-snowflake#1650] [Q&A] [answered]
I'm using VCS using Azure DevOps repos and want to switch roles by overriding the environment variable (SNOWFLAKE_ROLE).
I want to use USERADMIN role for USER creation and SYSADMIN role for Databases.

What is the best way to do this if I've configured VCS with my Terraform Cloud.

**Comment by tamathew** [comment, marked as answer]
I solved the problem by not declaring the SNOWFLAKE_ROLE as my environment or TF variable. Instead , I used used provider config in every single terraform file with an alias . The provider config would switch the role .

eg :  
filename : roles.tf

provider "snowflake" {

  role      = "USERADMIN"
  alias = "useradmin"

}

resource "snowflake_role" "role" {
  provider = snowflake.useradmin
  name = "DEVELOPER_ROLE"
  comment = "Developer role for Service accounts"
}


filename : databases.tf

provider "snowflake" {

  role      = "SYSADMIN"
  alias = "sysadmin"

}

resource "snowflake_database" "database" {
  provider = snowflake.sysadmin
  name = "SALES_DB"
  comment = "Sales database"
}

**Reply by culpgrant** [reply]
This is how I did it too!

**Comment by tamathew** [comment]
Resolved
